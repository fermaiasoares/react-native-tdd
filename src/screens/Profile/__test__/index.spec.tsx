import React from 'react';
import { render } from '@testing-library/react-native';

import { Profile } from '..';

describe('Profile Screen', () => {
    it('Check if profil`s screen is equal snapshot', () => {
        const profileScreen = render(<Profile />);
        expect(profileScreen).toMatchSnapshot();
    })

    it('Check if showing correctly user input name placeholder', () => {
        const { getByPlaceholderText } = render(<Profile />);
        const inputName = getByPlaceholderText('Nome');
        expect(inputName.props.placeholder).toBeTruthy();
    })

    it('Check if showing correctly user input last name placeholder', () => {
        const { getByPlaceholderText } = render(<Profile />);
        const inputName = getByPlaceholderText('Sobrenome');
        expect(inputName.props.placeholder).toBeTruthy();
    })

    it('Check if showing correctly save button text', () => {
        const { getByTestId } = render(<Profile />);
        const saveButton = getByTestId('btn-salvar');
        expect(saveButton).toBeTruthy();
    })
})