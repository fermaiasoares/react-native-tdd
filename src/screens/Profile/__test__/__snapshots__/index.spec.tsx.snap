// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Screen Check if profil\`s screen is equal snapshot 1`] = `
<View>
  <Text>
    Profile
  </Text>
  <TextInput
    autoCorrect={false}
    placeholder="Nome"
  />
  <TextInput
    autoCorrect={false}
    placeholder="Sobrenome"
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="btn-salvar"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Salvar
      </Text>
    </View>
  </View>
</View>
`;
